### PR TITLE
RUN-3386 null checks to avoid exception

### PR DESCRIPTION
**Pull request by pbaize**
This adds the null checks to avoid the runtime exception when center clicking. It does not resolve the issue of center clicking opening a blank window.

### CODE REVIEW

**Comment by weiteho** [comment]
this hides the exception but not really solved the underlying issue. We should figure out what to do with center click/ctrl + click and create corresponding openfin window objects. This way we don't even need to do null check here.

**Comment by pbaize** [comment]
Agreed, although we should leave the null checks in considering we sometimes set these variables to null.

File: src/browser/web_request_handler.ts [hunk 1]
```diff
@@ -84,10 +84,10 @@ function beforeSendHeadersHandler(details: RequestDetails, callback: (response:
         if (wc) {
             electronApp.vlog(1, `${moduleName}:beforeSendHeadersHandler got webcontents ${wc.id}`);
             const bw = wc.getOwnerBrowserWindow();
-            if (bw) {
+            if (bw && bw.id) {
```
Comment (HarsimranSingh) [inline src/browser/web_request_handler.ts]: Is there ever a chance that the BrowserWindow id is going to be `0`? Maybe safer to check `typeof`

**Comment by wenjunche** [comment]
Manual tests on disableModifiedClicks ?

File: src/renderer/api-decorator.js [hunk 2]
```diff
@@ -333,6 +333,25 @@ limitations under the License.
...
+            const mightOpenNewWindow = tag === 'A' || tag === 'IMG';
+            if (mightOpenNewWindow && modifiedClick) {
+                e.preventDefault();
+            } else if (modifiedClick && (tag === 'BUTTON' || tag === 'INPUT')) {
+                if (e.target.type === 'submit') {
+                    e.preventDefault();
+                }
+
+            }
+        });
```
Comment (whyn07m3) [inline src/renderer/api-decorator.js]: Code looks good as it is, just a refactored suggestion below
```
const { tagName, type } = e.target;
const modifiedClick = e.shiftKey || e.metaKey || e.ctrlKey || e.altKey;
const shouldPreventDefault =
    tagName === 'A' || tagName === 'IMG'
    ||
    type === 'submit' && (tagName === 'BUTTON' || tagName === 'INPUT');

if (modifiedClick && shouldPreventDefault) {
    e.preventDefault();
}
```

**Comment by pbaize** [comment]
Test results: https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a6a3dda3f32861e82b77c84

**Pull request merged**